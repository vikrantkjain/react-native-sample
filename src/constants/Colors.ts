/**
 * Declare all the color codes here for using in throughout the app.
 */

export const WHITE = '#FFFFFF';
export const BLACK = '#000000';
export const LIGHT_GRAY_BORDER = '#EDEDED';
export const LIGHT_GRAY = '#E4E4E4';
export const TRANSPARENT = '#00000000';
export const ALBASTAR_GRAY = '#F8F8F8';
export const CERULEAN = '#048de0';
export const AZURE_BLUE = 'blue';
export const DARK_CHARCOAL = '#333333';
export const DOVE_GRAY = '#333333';
export const DIM_GRAY = 'lightgrey';
export const GRAY = 'grey';
export const GREEN = 'green';
export const RED = 'red';
export const MAIN_COLOR = '#100A55';
export const GUN_SMOKE = '#808494';
export const GHOST_WHITE = '#F7F7FD';
export const PLATINUM = '#E0DEF7';
export const BLUE = '#7065F0';
export const LIGHT_BORDER = '#D6DDEB';
export const DARK_MAIN = '#000929';

/**
 * @use
 * import {colors} from 'something';
 * colors.EXAMPLE_COLOR
 */
export const colors = {
  EXAMPLE_COLOR: 'yellow',
};
