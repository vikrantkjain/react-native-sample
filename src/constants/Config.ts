/**
 * All configuration example
 * google api key, agora api key etc.
 */

import {Platform} from 'react-native';
import RNConfig from 'react-native-config';

export const appInfo = {
  name: 'DemoCode',
  mainDomain: RNConfig.API_URL,
  url: Platform.select({
    android:
      'https://play.google.com/store/apps/details?id=com.netflix.mediaclient',
    ios: 'https://apps.apple.com/in/app/netflix/id363590051',
  }),
};
