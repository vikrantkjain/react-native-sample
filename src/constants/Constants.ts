/**
 * Declare all the constant keys and their values here for using in throughout the app.
 */

import {Platform} from 'react-native';

export const LocalStorageKeys = {
  SHARED_PREFRENCE: 'DemoCode_pref',
  KEYCHAIN_SERVICE: 'DemoCode_service',
  ACCESS_TOKEN: 'access_token',
  DEVICE_TOKEN: 'device_token',
  USER_DATA: 'user_data',
  IS_INTRO_SHOW: 'is_intro_show',
  IS_KEYCHAIN_HAS_DATA: 'is_keychain_has_data',
};
export const ApiConstants = {
  APP_VERSION: '1.0.0',
  DEVICE_TYPE: Platform.OS,
  CERTIFICATE_TYPE: __DEV__ ? 'development' : 'distribution',
  GET: 'get',
  POST: 'post',
  POST_WITH_FORM: 'post_with_form',
  PUT: 'put',
  DELETE: 'delete',
  SERVER_DATE_TIME_FORMAT: 'YYYY-MM-DD HH:mm:ss',
};
export const AppConstants = {
  IS_LOG: true,
  AUTH_TOOLBAR: 'AUTH_TOOLBAR',
  NETWORK_CHECK: false,
  DEVICE_TYPE: Platform.OS,
  APP_VERSION: '1.0.0',
  TOUCH_RADIUS: 20,
  TOUCH_OPACITY: 0.5,
  TOUCH_DURATION: 500,
  MALE: 'male',
  FEMALE: 'female',
  EMAIL_REGEX: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,8})+$/,
  MOBILE_REGEX: /^[0][1-9]\d{9}$|^[1-9]\d{5,15}$/,
  NUMBER_OR_DECIMAL_REGEX: /^(\d*\.)?\d+$/,
  NUMBER_REGEX: /^\d+$/,
  CHARACTER_OR_NUMBER_REGEX: /^(?=.*[A-Z])(?=.*\d)[A-Z\d]{4,7}$/,
  NAME_REGEX: /^[^\s].+[a-zA-Z]+[a-zA-Z]+$/,
  FULLNAME_REGEX: /^[a-zA-z]+([\s][a-zA-Z]+)*$/,
  PASSWORD_REGEX: /^(?=.*[a-z])(?=.*[a-z])(?=.*\d)\S{8,18}$/,
  PIN_CODE_REGEX: /^[1-9][0-9]{5}$/,
  ADDRESS_REGEX: /^[a-zA-Z0-9\s,.'-]{3,}$/,
  DATE_FORMAT_APP: 'DD MMM,YYYY',
  TIME_FORMAT_APP: 'HH:MM A',
  SUBJECT_MAX_LENGTH: 100,
  MESSAGE_MAX_LENGTH: 1000,
  DEV: __DEV__,
};
export const StoreConstantValues = {
  USER_ACCESS_TOKEN: '',
  USER_INFO_DATA: '',
  USER_ID: '',
  AUTHORIZATION_DATA: '',
};
