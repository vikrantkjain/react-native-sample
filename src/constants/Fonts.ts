/**
 * Declare all the fonts name and size here for using in throughout the app.
 */

export const FontSize = {
  /** h1 = 30 */
  h1: 30,
  /** h2 = 24 */
  h2: 24,
  /** h3 = 22 */
  h3: 22,
  /** h4 = 20 */
  h4: 20,
  /** h5 = 18 */
  h5: 18,
  /** h6 = 16 */
  h6: 16,

  /** t1 = 14 */
  t1: 14,
  /** t2 = 12 */
  t2: 12,
  /** t3 = 10 */
  t3: 10,
  /** t4 = 8 */
  t4: 8,
};

export const FontFamily = {
  regular: 'PlusJakartaSans-Regular',
  bold: 'PlusJakartaSans-Bold',
  semi_bold: 'PlusJakartaSans-SemiBold',
  medium: 'PlusJakartaSans-Medium',
};
