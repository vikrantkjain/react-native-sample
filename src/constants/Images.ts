/**
 * Declare all the image name and path here for using in throughout the app.
 */
export const Images = {
  ic_check: require('../assets/images/ic_check.png'),
  ic_drop_down: require('../assets/images/ic_drop_down.png'),
  ic_eye_visible: require('../assets/images/ic_eye_visible.png'),
  ic_eye_hidden: require('../assets/images/ic_eye_hidden.png'),
  ic_green_star: require('../assets/images/ic_green_star.png'),
  ic_grey_star: require('../assets/images/ic_grey_star.png'),
  ic_hal_star_rating_yellow: require('../assets/images/ic_hal_star_rating_yellow.png'),
  ic_half_star_green_rating: require('../assets/images/ic_half_star_green_rating.png'),
  ic_yellow_star: require('../assets/images/ic_yellow_star.png'),
  ic_uncheck: require('../assets/images/ic_uncheck.png'),
  ic_left: require('../assets/images/ic_left.png'),
  ic_star: require('../assets/images/ic_star.png'),
  ic_minus: require('../assets/images/ic_minus.png'),
  ic_plus: require('../assets/images/ic_plus.png'),
  ic_drawer: require('../assets/icons/ic_drawer.png'),
  ic_user: require('../assets/icons/ic_user.png'),
  ic_search: require('../assets/icons/ic_search.png'),
  ic_home: require('../assets/icons/ic_home.png'),
  ic_history: require('../assets/icons/ic_history.png'),
  ic_menu: require('../assets/icons/ic_menu.png'),
  ic_back: require('../assets/icons/ic_back.png'),
  ic_back_black: require('../assets/icons/ic_back_black.png'),
  ic_rotate_Camera: require('../assets/images/rotateCamera.png'),
  ic_flash: require('../assets/images/flash.png'),
  ic_cross: require('../assets/images/ic_minus.png'),
  ic_demoCodeLogo: require('../assets/icons/ic_demoCodeLogo.png'),
  ic_google: require('../assets/icons/ic_google.png'),
  ic_logout: require('../assets/icons/ic_logout.png'),
};

export default Images;
