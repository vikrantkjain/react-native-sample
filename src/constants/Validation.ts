/**
 * Declare all the validation strings here for using in throughout the app.
 */
export const Validation = {
  appName: 'React Native template',
  validFirstNameFormat: 'Please enter valid first name',
  validLastNameFormat: 'lease enter valid last name',
  validLastName: 'Please enter valid last name',
  validEmail: 'Please enter email Id',
  validEmailFormat: 'Please enter a valid email Id',
  validPassword: 'Please enter password',
  validOldPassword: 'Please enter old password',
  validNewPassword: 'Please enter new password',
  validPasswordDetail:
    'Password must contain atleast 8 characters, 1 special character, 1 uppercase & 1 number',
  validConfirmPassword: 'Please enter confirm password',
  validConfirmPasswordLength: 'Confirm Password and Password does not match',
  validConfirmPasswordNotMatch:
    'New password and confirm password does not match',
  validConfirmNewPassword: 'Please enter confirm new password',
  forgotPwdTitle: 'Forgot Password?',
  forgotPwdSubTitle: 'Please enter your email id to reset your \n password.',
  emailId: 'Email ID',
  sendVerificationCode: 'Send Verification Code',
  FirstNameError: 'Please enter your full name',
  validFirstName: 'Please enter valid name',
  exceedNameLimit: 'Please enter less than 20 characters',
  MobileNumberError: 'Please enter mobile number',
  validMobileNumber: 'Please enter a valid mobile number',
  validMobileFormat: 'Please enter a valid mobile number',
  validState: 'Please enter state',
  validStateFormat: 'Please enter valid state',
  verifyCode: 'Verify Code',
  validOneTimePassword: 'Please enter OTP',
  validOneTimeLength: 'Please enter 4 digit valid OTP',
  validAdd: '',
  validAddFormat: '',
  validPincode: '',
  validPincodeFormat: '',
  validCity: '',
  validLocalityFormat: '',
  validCityFormat: '',
  validEmailMobileEmail: '',
  validOneTimePwd: '',
};
