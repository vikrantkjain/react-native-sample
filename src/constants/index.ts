export * from './Fonts';
export * from './Images';
export * from './Colors';
export * from './Strings';
export * from './Constants';
export * from './Config';
export * from './Validation';

export const MonthNames = {
  '01': 'Jan',
  '02': 'Feb',
  '03': 'Mar',
  '04': 'Apr',
  '05': 'May',
  '06': 'Jun',
  '07': 'July',
  '08': 'Aug',
  '09': 'Sep',
  10: 'Oct',
  11: 'Nov',
  12: 'Dec',
};
