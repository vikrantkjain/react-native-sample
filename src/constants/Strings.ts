import {appInfo} from './Config';

/**
 * Declare all the strings here for using in throughout the app.
 */
export const Strings = {
  pleaseWait: 'Please wait...',
  noDataFound: 'No data found...',
  appUpdateDesc: `Update to the latest version to continue using ${appInfo.name}`,
  appUpdateTitle: 'Keep up-to-date',
  updateLater: 'Update Later',
  updateNow: 'Update Now',
  noInternet: 'You are offline. Please check your internet connection',
  backOnline: 'You are back online',
  noInternetConnection: 'No internet connection.',
  somethingPleaseTryAgain: 'Some error occurred please try again',
  pleaseTryAgain: 'Please try again after some time.',
  internalServer: 'Internal Server Error',
  noData: 'Not found',
  forgotPwdTitle: 'Forgot Password?',
  forgotPwdSubTitle: 'Please enter your email id to reset your \n password.',
  sendVerificationCode: 'Send Verification Code',
  email: 'Email',
  name: 'Name',
  mobileNumber: 'Mobile Number',
  password: 'Password',
  confirmPassword: 'Confirm Password',
  signup: 'Sign up for free',
  alreadyHaveAnAccount: 'Already have an account?',
  signIn: 'SignIn',
  login: 'Login',
  createAccount: "Don't have an account?",
  verifyCode: 'Verify Code',
  ok: 'Ok',
  networkError: 'Network error',
  title: 'Democode',
  wb: 'Welcome back',
  wbMessage: 'Welcome back! Please enter your details',
  forgotPassword: 'Forgot Password?',
  continueWithGoogle: 'Continue with Google',
  signUpText: 'Sign up',
  firstName: 'First name',
  lastName: 'Last name',
  signUpSuccessfully: 'SignUp successfully',
  areYouSure: 'Are you sure you want to logout?',
  notAvailable: 'This feature is currently not available in the demo code',
};
