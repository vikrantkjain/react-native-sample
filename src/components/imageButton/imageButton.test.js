import renderer from 'react-test-renderer';
import ImageButton from './index';
import {Images} from '../../constants';
const url =
  'https://cdn.kinandcarta.com/-/media-assets/images/kincarta/insights/2022/02/react-native/react_hero.png?as=0&iar=0&w=1200&rev=61e1dad3af7e465e9544cf8490237772&hash=0AD31383BCBA1DA1C88546327312BA33';

jest.mock('react-native-config', () => 'RNConfig');
test('renders correctly', () => {
  const tree = renderer
    .create(<ImageButton source={Images.ic_back} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
