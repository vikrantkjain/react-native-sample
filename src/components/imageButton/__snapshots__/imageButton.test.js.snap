// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "testUri": "../../../src/assets/icons/ic_back.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
